meta: refine contribution guide (#51)
botlabs-gg/yagpdb-docs-v2: .github/CONTRIBUTING.md

# Contributing Guidelines

## Got a Question?

> Please do not open issues for general support such as "how to do reaction roles"—read the docs and/or ask on the
> [support server] for that.
> 
> [support server]: https://discord.gg/4udtcA5
> 
> To keep this tracker clean, we will systematically close issues regarding general support and redirect to the
> aforementioned support server.

Reviewer: IMO, we should try to keep tangential warnings directed to confused users to a minimum here. They aren't likely to read this document anyway :-)
```suggestion
We welcome suggestions for improving the documentation on our issue tracker—outdated information, missing items, typos, or clarity issues are all relevant.

General support queries and possible bugs in the YAGPDB bot are **not** in scope and will be closed without comment; use the [support server] instead.

[support server]: https://discord.gg/4udtcA5
```

Author: Yet Another General Purpose Discord Bot bot ah yes

## Found a Bug?

Do **not** use this tracker to file maybe-bugs in YAGPDB. We have a dedicated channel on the support server for that, to
which you can post after it was confirmed to be one.

Should there be general issues with the documentation (grammatical structure, general flow, clarification needed), do

> feel free to open an issue clearly outlining what is wrong about it and consider suggesting how to fix it.

Reviewer: combine with above

## Submission Guide

Thanks for contributing to the YAGPDB documentation! Please take a moment to review this document in order to make the
contribution process easy and effective for everyone involved.

### Requirements

> You will at the very least need an [EditorConfig]-capable (text) editor like VSCode or Neovim. Though the GitHub web
> interface may work fine, you cannot save in-between without having to commit your changes, which is not ideal.

Reviewer: ```suggestion
Small changes can be authored directly using the GitHub web interface, though we highly recommend an [EditorConfig]-capable editor like VS Code or Neovim for any extended work.
```
"need an editor" and "the GitHub web interface may work fine" are slightly contradictory :-)

without our custom syntax highlighting. If you prefer to view the site as it would appear on production, run the
following commands instead:

```shellsession
$ npm run build && npm run preview
```

This will build the site, execute the highlighting post-processor, and run a local server on `http://localhost:4173`.

#### Committing Your Changes

Once you are happy with your changes and ensured that the page builds without errors, you can commit your changes. We
have a small set of rules for commit messages, which help both you and us keep track of changes more easily.

A commit message consists of a header, a body, and a footer:

```txt
<header>
<blank line>
<body>
<blank line>
<footer>
```

The header is mandatory and should ideally be limited to about 50 characters. GitHub will automatically break the header
line once it exceeds 72 characters, so consider that the hard limit. If your change is trivial enough, like a typo or
fixing a broken link, the header is all you need. It should be written in the imperative mood and be descriptive of what
precisely was changed.

```diff
- Update README.md
+ readme: fix broken link to support server
```

> If you find yourself struggling to concisely summarize your changes in the header, you might want to consider splitting
> them up into separate, more atomic commits, perhaps even separate PRs if these changes can be reasonably separated.

Reviewer: ```suggestion
If you find yourself struggling to concisely summarize your changes in the header, consider splitting them up into multiple commits or perhaps even multiple PRs if appropriate.
```

Author: oh the irony that i myself struggled to concisely author this paragraph
